### PR TITLE
feat(dotnet): Support for JSII_DEBUG and JSII_RUNTIME

### PR DESCRIPTION
**Pull request by assyadh**
Fixes https://github.com/aws/jsii/issues/438

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs [hunk 1]
```diff
@@ -9,25 +9,36 @@ public class NodeProcess : INodeProcess
...
-                    Arguments = "--max-old-space-size=4096 " + jsiiRuntimeProvider.JsiiRuntimePath,
+                    Arguments = "--max-old-space-size=4096 " + runtimePath,
                     RedirectStandardInput = true,
                     RedirectStandardOutput = true,
                     RedirectStandardError = true
                 }
             };
 
-            _process.StartInfo.EnvironmentVariables.Add("JSII_AGENT", "DotNet/" + Environment.Version);
+            _process.StartInfo.EnvironmentVariables.Add(JsiiAgent, "DotNet/" + Environment.Version);
```
Comment (RomainMuller) [inline packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs]: Any chance we can be more "accurate" than "DotNet"? Is it possible to get ideas of the framework build version or something?

Comment (assyadh) [reply packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs]: Yup, we do it in the SDK, will do the same here.

File: packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs [hunk 2]
```diff
@@ -48,5 +64,23 @@ void IDisposable.Dispose()
...
+        /// and item2 is the assembly file version (ie 1.0.0.0)</returns>
+        private Tuple<string, string> GetAssemblyFileVersion()
+        {
+            var assembly = typeof(NodeProcess).GetTypeInfo().Assembly;
+            var assemblyFileVersionAttribute = assembly.GetCustomAttribute(typeof(AssemblyFileVersionAttribute)) as AssemblyFileVersionAttribute;
+            var frameworkAttribute = assembly.GetCustomAttribute(typeof(TargetFrameworkAttribute)) as TargetFrameworkAttribute;
+            return new Tuple<string, string>(
+                frameworkAttribute == null ? "Unknown" : frameworkAttribute.FrameworkName,
+                assemblyFileVersionAttribute == null ? "Unknown" : assemblyFileVersionAttribute.Version
+                );
```
Comment (RomainMuller) [inline packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs]: Is this bracket intended to be indented there? Looks to me it should be indented one less deep?

File: packages/jsii-dotnet-runtime-test/test/Amazon.JSII.Runtime.IntegrationTests/ComplianceTests.cs [hunk 3]
```diff
@@ -915,7 +915,7 @@ public void OptionalAndVariadicArgumentsTest()
         [Fact(DisplayName = Prefix + nameof(JsiiAgent))]
         public void JsiiAgent()
         {
-            Assert.Equal("DotNet/" + Environment.Version.ToString(), JsiiAgent_.JsiiAgent);
+            Assert.Equal("DotNet/" + Environment.Version.ToString() + "/.NETStandard,Version=v2.0/1.0.0.0", JsiiAgent_.JsiiAgent);
```
Comment (RomainMuller) [inline packages/jsii-dotnet-runtime-test/test/Amazon.JSII.Runtime.IntegrationTests/ComplianceTests.cs]: What does it take for this to break? Does it need manual action (source change somewhere else)? Or can it change just by getting updated toolkit?

Comment (assyadh) [reply packages/jsii-dotnet-runtime-test/test/Amazon.JSII.Runtime.IntegrationTests/ComplianceTests.cs]: An update to your local framework.

I could just test that it StartsWith("DotNet")?

File: packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs (hunk 2 above)
Comment (assyadh) [reply packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs]: Yeah, rider bug. Indentation is wrong when a new is in a return statement for some reason.

Fixed.

File: packages/jsii-dotnet-runtime-test/test/Amazon.JSII.Runtime.IntegrationTests/ComplianceTests.cs (hunk 3 above)
Comment (costleya) [reply packages/jsii-dotnet-runtime-test/test/Amazon.JSII.Runtime.IntegrationTests/ComplianceTests.cs]: As an aside, were we not going to change to core 2.1 target?

Standard is great, except it makes it look like we support framework - and I don't even really want to give developers that option.

File: packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs [hunk 4]
```diff
@@ -48,5 +64,23 @@ void IDisposable.Dispose()
...
+        /// <returns>A tuple where Item1 is the target framework
+        /// ie .NETStandard,Version=v2.0
+        /// and item2 is the assembly file version (ie 1.0.0.0)</returns>
+        private Tuple<string, string> GetAssemblyFileVersion()
+        {
+            var assembly = typeof(NodeProcess).GetTypeInfo().Assembly;
+            var assemblyFileVersionAttribute = assembly.GetCustomAttribute(typeof(AssemblyFileVersionAttribute)) as AssemblyFileVersionAttribute;
+            var frameworkAttribute = assembly.GetCustomAttribute(typeof(TargetFrameworkAttribute)) as TargetFrameworkAttribute;
+            return new Tuple<string, string>(
+                frameworkAttribute == null ? "Unknown" : frameworkAttribute.FrameworkName,
```
Comment (costleya) [inline packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs]: frameworkAttribute?.FrameworkName ?? "Unknown"

:P This is just a nit.

**Pull request merged**